lib/repl.js

```diff
@@ -689,8 +690,31 @@ function intFilter(item) {
   return /^[A-Za-z_$]/.test(item);
 }
 
+const DEFAULT_PROPERTIES = {
+  ARRAY: Object.getOwnPropertyNames([]).filter(intFilter),
+  BUFFER: Object.getOwnPropertyNames(Buffer.alloc(1)).filter(intFilter)
+};
+
+function mayBeLargeObject(obj) {
+  return (Array.isArray(obj) || Buffer.isBuffer(obj));
+}
+
 function filteredOwnPropertyNames(obj) {
   if (!obj) return [];
+  if (mayBeLargeObject(obj) && obj.length > 1e6) {
+    this._writeToOutput('\r\n');
+    process.emitWarning(
+      'Instance is too large that the completion may missing ' +
```

Comment:
I feel like "so" might be better than "that" here (but I'm not a native speaker).